bundles/org.openhab.binding.salus/pom.xml

```diff
@@ -34,6 +40,168 @@
       <scope>compile</scope>
     </dependency>
     <!-- END caffeine -->
+    <!-- START AWS -->
```

Comment:
Would like to have an extra set of eyes regarding this list of dependency's @openhab/add-ons-maintainers. 
Same for the bnd.importpackage.

Reply:
@openhab/add-ons-maintainers

Reply:
haha, hope you are not trolling.
It is a group that receives a ping when added like this to a post. So give them some time to respond.

Reply:
but shouldn't @openhab/add-ons-maintainers change to a linkish thing?

Reply:
It is. 
![image](https://github.com/openhab/openhab-addons/assets/13950897/2973ca18-3900-480a-a9db-6bb1f282aebf)

Reply:
:o 

![obraz](https://github.com/openhab/openhab-addons/assets/2962846/a778da8d-e67a-4a0a-9e39-0bfe853de0d9)

Reply:
@wborn if i remember correct i have seen you looking at the AWS dependencies before. To me this looks valid, but would like your comment too.

Reply:
Yes we decided to not use these cloud SDKs if possible because they significantly add bloat to addons making them huge.

See:
* https://github.com/openhab/openhab-addons/pull/14669
* https://github.com/openhab/openhab-addons/pull/14770

Reply:
It's gonna be hard... I don't want to implement my own versions of signing algorithms or auth with cognito ;(

Reply:
The Windcentrale Binding also uses Cognito without the SDK.

Reply:
🤣🤣🤣

I just open first class in windcentrale and I found that he copied the same class as I did 🤣🤣🤣

![obraz](https://github.com/openhab/openhab-addons/assets/2962846/f69d6d52-922f-4cc6-a398-8050961bb188)

Although I think he added some more methods there so I will take a look on them

Reply:
@lsiepel  can  I copy [`AuthenticationHelper`](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/api/AuthenticationHelper.java) from windcentrale into my project?

Reply:
> Although I think he added some more methods there so I will take a look on them

He is me. Ofcourse I had to clean up the code and use more recent Java constructs. :wink:

Reply:
> can I copy [AuthenticationHelper](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/api/AuthenticationHelper.java) from windcentrale into my project?

If such tiny duplications allow for removing all the AWS Cloud SDK dependencies that would be preferred IMHO. :+1:

Reply:
@wborn  what about [signing request](https://github.com/openhab/openhab-addons/blob/968a50be5d489e018887bef4d5cc328216b403ad/bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/aws/http/AwsSalusApi.java#L162)? Are you doing it also? I cannot find proper code

Reply:
@lsiepel I've removed cognito deps and only left one for signing. I'm not keen on rewriting signing on my own

Reply:
Much better, thanks

Reply:
I don't think I've used the signing
Maybe https://stackoverflow.com/a/62144380/20718063 helps?

That `aws-crt` dependency is huge at ~17MB.
It contains native libraries for many OSes and architectures. :face_with_diagonal_mouth:

Reply:
@lsiepel, @magx2 - I think you may have missed @wborn's comment about the `aws-crt` dependency? The windows snapshot `openhab-4.2.0-SNAPSHOT.zip` is today 106 MB in total (build 4110). It has been approximately this size since around October 2023 (in September it was 102 MB). I guess we'll see tomorrow if the size will increase significantly.

Reply:
@magx2 are you able to come up with a follow up PR? 
Most recent files: https://github.com/frusal/java-amazon-request-signature-v4

Reply:
@lsiepel from the doc:

> For now, I would suggest you just copy its content straight into your source code, if there will be any demand, I might publish it on [maven central](https://mvnrepository.com/repos/central).

should I do it?

Reply:
If yes then should I put it in original package `com.frusal.amazonsig4` or in my package?

Reply:
moved to https://github.com/openhab/openhab-addons/pull/16840

Reply:
> The Windows snapshot `openhab-4.2.0-SNAPSHOT.zip` is today 106 MB

But doesn't include add-ons. 😳 Instead I had a look at before/after:
- https://openhab.jfrog.io/ui/builds/openHAB-Addons/1406/1717212309812/published/org.openhab.addons.bundles:org.openhab.binding.salus:4.2.0-SNAPSHOT
- https://openhab.jfrog.io/ui/builds/openHAB-Addons/1407/1717385106932/published/org.openhab.addons.bundles:org.openhab.binding.salus:4.2.0-SNAPSHOT

1,12 MB vs. 17,6 MB, so exactly like @wborn predicted.